test/terra/backends/aer_simulator/test_wrapper_unitary_simulator.py

```diff
@@ -13,80 +13,100 @@
+        result = backend.run(circuits, shots=1).result()
         targets = ref_unitary_gate.unitary_gate_unitary_deterministic()
-        job = execute(circuits,
-                      self.SIMULATOR,
-                      shots=1,
-                      **self.BACKEND_OPTS)
-        result = job.result()
         self.assertSuccess(result)
         self.compare_unitary(result, circuits, targets)
 
-    def test_unitary_gate_circuit_run(self):
+    @supported_devices
+    def test_unitary_gate_circuit_run(self, device):
         """Test simulation with unitary gate circuit instructions."""
+        backend = self.backend(device=device)
         circuits = ref_unitary_gate.unitary_gate_circuits_deterministic(
             final_measure=False)
+        circuits = transpile(circuits, backend, optimization_level=1)
+        result = backend.run(circuits, shots=1).result()
         targets = ref_unitary_gate.unitary_gate_unitary_deterministic()
-        tqc = transpile(circuits,self.SIMULATOR)
-        job = self.SIMULATOR.run(tqc, shots=1, **self.BACKEND_OPTS)
-        result = job.result()
         self.assertSuccess(result)
         self.compare_unitary(result, circuits, targets)
 
-    def test_diagonal_gate(self):
+    @supported_devices
+    def test_diagonal_gate(self, device):
         """Test simulation with diagonal gate circuit instructions."""
+        backend = self.backend(device=device)
         circuits = ref_diagonal_gate.diagonal_gate_circuits_deterministic(
             final_measure=False)
+        circuits = transpile(circuits, backend, optimization_level=1)
+        result = backend.run(circuits, shots=1).result()
         targets = ref_diagonal_gate.diagonal_gate_unitary_deterministic()
-        job = execute(circuits,
-                      self.SIMULATOR,
-                      shots=1,
-                      **self.BACKEND_OPTS)
-        result = job.result()
         self.assertSuccess(result)
         self.compare_unitary(result, circuits, targets)
 
     # ---------------------------------------------------------------------
     # Test global phase
     # ---------------------------------------------------------------------
 
-    def test_qobj_global_phase(self):
+    @supported_devices
+    def test_qobj_global_phase(self, device):
         """Test qobj global phase."""
-
+        backend = self.backend(device=device)
         circuits = ref_1q_clifford.h_gate_circuits_nondeterministic(
             final_measure=False)
+        circuits = transpile(circuits, backend, optimization_level=1)
+        result = backend.run(circuits, shots=1).result()
         targets = ref_1q_clifford.h_gate_unitary_nondeterministic()
-
-        qobj = assemble(transpile(circuits, self.SIMULATOR),
-                        shots=1, **self.BACKEND_OPTS)
-        # Set global phases
-        for i, _ in enumerate(circuits):
+        for i, circuit in enumerate(circuits):
             global_phase = (-1) ** i * (pi / 4)
```

Comment:
Unrelated, but wow did this line throw me.  I read it as `(-1)**1j * (pi / 4)` which was _really_ confusing me.